modules/statistics/php/stats_demographic.class.inc

```diff
@@ -163,65 +163,73 @@ class Stats_Demographic extends \NDB_Form
         foreach (\Utility::getProjectList() as $key => $value) {
             $projects[$key] = $value;
         }
-        $currentProject = null;
-        if ($_REQUEST['DemographicProject'] ?? ''
-        ) {
-            $currentProject = htmlspecialchars($_REQUEST['DemographicProject']);
+
+        $demographicProject = $_REQUEST['DemographicProject'] ?? null;
+        $Param_Project      = '';
+
+        if (!empty($demographicProject)) {
+            $demographicProject = htmlspecialchars($demographicProject);
             $this->tpl_data['CurrentProject']
                 = array(
-                   'ID'   => $currentProject,
-                   'Name' => $projects[$currentProject],
+                   'ID'   => $demographicProject,
+                   'Name' => $projects[$demographicProject] ?? '',
                   );
             $Param_Project       ='AND (c.ProjectID IS NULL OR c.ProjectID=:pid) ';
-            $this->params['pid'] = htmlspecialchars($_REQUEST['DemographicProject']);
-        } else {
-            $Param_Project ='';
+            $this->params['pid'] = $demographicProject;
         }
 
         //SUBPROJECTS
-        $subprojList ="";
-        $subprojects = \Utility::getSubprojectsForProject($currentProject);
-        foreach ($subprojects as $key=>$val) {
-            $subprojList .= $key.",";
-        }
-        $subprojList = substr($subprojList, 0, -1);
-        if (!empty($subprojList)) {
-            $suproject_query ="AND s.SubprojectID IN ($subprojList)";
-        } else {
-            $suproject_query ='';
+        $subprojList      = '';
+        $subproject_query = '';
+        $Visits           = \Utility::getExistingVisitLabels($demographicProject);
+        $project          = $demographicProject != '' ? null : $demographicProject;
+        $subprojects      = \Utility::getSubprojectsForProject($project);
+
+        if (!empty($subprojects)) {
+            foreach ($subprojects as $key=>$val) {
```

Comment:
If `$val` is not used, just do `foreach ($subprojects as $subproject) {`

In fact, this section may be better written as

> $subproject_list = implode(',', $subprojects);

That way you can also remove line 192 because it seems to be chopping off a comma that shouldn't be there anyway...